server.js

```diff
@@ -0,0 +1,49 @@
+const { response } = require("express")
+const express = require("express")
+const app = express()
+const cors = require("cors")
+app.use(cors())
+
+let users = [
+    { _id: "u1", username: "user1", password: "pw1" },
+    { _id: "u2", username: "user2", password: "pw2" },
+    { _id: "u3", username: "user3", password: "pw3" },
+]
+
+app.use(express.json()) //parse all incoming body data 
+
+app.listen(5000, () => {
+    console.log("API has started successfully on PORT 5000")
+})
+
+
+app.get("/users", (req, res) => {
+    res.json(users)
+})
+
+app.post("/signup", (req, res) => {
+    console.log(req.body)
+    res.json(req.body)
```

Comment:
Slight issue here, you return the response too early with res.json.

Usually returning the response to the browser is the LAST line in the route, after we did all other stuff.

In POST routes the convention is to return the created item, including the ID. 

So actually you can fix it easily, by moving the res.json line to the end and return the newly created user:
`res.json( newUser )`. And all will be fine :)